tests_new/integration_tests/README.md

```diff
@@ -12,59 +12,73 @@ Currently **we support only Cognito based deployments** but support for any IdP
-          "password": "...",
-          "groups": [
-            "testGroup1"
-          ]
-        },
-        "testUser2": {
-          "username": "testUser2",
-          "password": "...",
-          "groups": [
-            "testGroup2"
-          ]
-        },
-        "testUser3": {
-          "username": "testUser3",
-          "password": "...",
-          "groups": [
-            "testGroup3"
-          ]
-        },
-        "testUser4": {
-          "username": "testUser4",
-          "password": "...",
-          "groups": [
-            "testGroup4"
-          ]
-        }
-      },
-      "envs": {
-        "session_env1": {
-          "accountId": "...",
-          "region": "eu-central-1"
-        },
-        "session_env2": {
-          "accountId": "...",
-          "region": "eu-west-1"
-        }
-      }
-    }
-    ```
+       - For this deployment the `config.json` flag `cdk_pivot_role_multiple_environments_same_account` must be set to `true` if an AWS account is going to be reused for multiple environments,
+         - Second test account is bootstraped, and first account is added to trusted policy in target regions
+          ```cdk bootstrap --trust <first-account-id> -c @aws-cdk/core:newStyleStackSynthesis=true --cloudformation-execution-policies arn:aws:iam::aws:policy/AdministratorAccess aws://<second-account-id>/region```
+         - AWS profile is set for second test account and the name is in testdata.
+           - An SSM parameter (`/dataall/{env_name}/testdata`) in the DEPLOYMENT ACCOUNT with the following contents
+               ```
+               {
+                 "users": {
+                   "testUserTenant": {
+                     "username": "testUserTenant",
+                     "password": "...",
+                     "groups": [
+                       "DAAdministrators"
+                     ]
+                   },
+                   "testUser1": {
+                     "username": "testUser1",
+                     "password": "...",
+                     "groups": [
+                       "testGroup1"
+                     ]
+                   },
+                   "testUser2": {
+                     "username": "testUser2",
+                     "password": "...",
+                     "groups": [
+                       "testGroup2"
+                     ]
+                   },
+                   "testUser3": {
+                     "username": "testUser3",
+                     "password": "...",
+                     "groups": [
+                       "testGroup3"
+                     ]
+                   },
+                   "testUser4": {
+                     "username": "testUser4",
+                     "password": "...",
+                     "groups": [
+                       "testGroup4"
+                     ]
+                   }
+                 },
+                 "aws_profiles": {
+                     "second": "second_int_test_profile"
+                 },
```

Comment:
I think the use of `aws_profiles` will not play very well with CodeBuild, instead I propose to use the existing infrastracture  (see `session_env1_aws_client`).

As discussed offline you need this account to test consumption roles. By using the integration test account you can solve it using the two following patterns
1. (simpler) add the already created (during env deployment) integration-test role directly as a consumption role. Current (CodeBuild/Local) account have already permissions to assume this role so you can use STS to assume it and then run S3 queries to make sure that the share was succesful.
2. (more complex) use the integration-test role to create new roles in the target account that you will register as consumption roles. Then proceed with assuming those roles and testing for S3 access.

Reply:
I implemeted via AssumeRole